be/btree.c

```diff
@@ -13159,6 +13170,261 @@ static void ut_btree_crc_persist_test_internal(struct m0_btree_type   *bt,
 	btree_ut_fini();
 }
 
+static void ut_mtree_mthread_test(void)
+{
+	int                          rc;
+	int                          ksize;
+	int                          vsize;
+	struct btree_ut_thread_info *ti;
+	int                          i;
+	int                          k;
+	int                          cur_arr_id = 0;
+	uint16_t                     cpu;
+	void                        *rnode;
+	struct m0_btree_op           b_op       = {};
+	struct btree_mthreads_data {
+		struct m0_btree_type   btree_type;
+		uint64_t               node_size;
+		uint64_t               no_of_trees;
+	} btrees_mthreads[] = {
+		{
+			{
+				M0_BT_BALLOC_GROUP_EXTENTS,
+				M0_MEMBER_SIZE(struct m0_ext, e_start),
+				M0_MEMBER_SIZE(struct m0_ext, e_end)
+			},
+			4096,
+			1
+		},
+		{
+			{
+				M0_BT_BALLOC_GROUP_DESC,
+				8,
+				96
+			},
+			4096,
+			1
+		},
+		{
+			{
+				M0_BT_EMAP_EM_MAPPING,
+				sizeof(struct m0_be_emap_key),
+				RANDOM_VALUE_SIZE
+			},
+			16384,
+			1
+		},
+		{
+			{
+				M0_BT_COB_NAMESPACE,
+				RANDOM_KEY_SIZE,
+				RANDOM_VALUE_SIZE
+			},
+			8192,
+			1
+		},
+		{
+			{
+				M0_BT_COB_OBJECT_INDEX,
+				sizeof(struct m0_cob_oikey),
+				RANDOM_VALUE_SIZE
+			},
+			8192,
+			1
+		},
+		{
+			{
+				M0_BT_COB_FILEATTR_BASIC,
+				sizeof(struct m0_cob_fabkey),
+				RANDOM_VALUE_SIZE
+			},
+			8192,
+			1
+		},
+		{
+			{
+				M0_BT_COB_FILEATTR_OMG,
+				sizeof(struct m0_cob_omgkey),
+				sizeof(struct m0_cob_omgrec)
+			},
+			8192,
+			1
+		},
+		{
+			{
+				M0_BT_COB_BYTECOUNT,
+				sizeof(struct m0_cob_bckey),
+				sizeof(struct m0_cob_bcrec)
+			},
+			8192,
+			1
+		}
```

Comment:
Add two cases atleast for cas tree type, refer:
`bt.ksize = -1; bt.vsize = -1;` and `bt.ksize = sizeof(struct fid_key); bt.vsize = sizeof(struct meta_value);`
This will increase coverage with possible bigger node sizes.

Reply:
> bt.ksize = sizeof(struct fid_key); bt.vsize = sizeof(struct meta_value);
sizeof fid_key is not multiple of 8 it is 25 bytes so seems difficult to include in our UT as we support only multiple of 8. 

```C
 102 struct fid_key {
 103         struct generic_key fk_gkey;
 104         struct m0_fid      fk_fid;
 105 };
```

```C
  84 struct generic_key {
  85         /* Actual length of gk_data array. */
  86         uint64_t gk_length;
  87         uint8_t  gk_data[0];
  88 };
```
```C
 38 struct m0_fid {
 39         uint64_t f_container;
 40         uint64_t f_key;
 41 }
```

Reply:
Done